openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py

```diff
@@ -1,22 +1,161 @@
 """PathstreamExternalEnrollment class file."""
 import logging
+import os
 from datetime import datetime
 
+import boto3
+from botocore.exceptions import ClientError
+from django.conf import settings
 from django.db import IntegrityError
 
 from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
 from openedx_external_enrollments.models import ExternalEnrollment
 
 LOG = logging.getLogger(__name__)
+COMPLETED = True
+UNCOMPLETED = False
+
+
+class S3NotInitialized(BaseException):
+    """Exception when _init_s3 has not been initialized before calling _upload_file or _download_file."""
+
+
+class ExecutionError(BaseException):
+    """Exception when the proccess of executing PathstreamExternalEnrollment.execute_upload fails."""
 
 
 class PathstreamExternalEnrollment(BaseExternalEnrollment):
     """
     PathstreamExternalEnrollment class.
     """
+    def __init__(self):
+        self.client = None
+        self.S3_BUCKET = settings.OEE_PATHSTREAM_S3_BUCKET
+        self.S3_FILE = settings.OEE_PATHSTREAM_S3_FILE
+
     def __str__(self):
         return 'pathstream'
 
+    def _init_s3(self):
+        """Start S3 connection.
+        """
+        self.client = boto3.client(
+            's3',
+            aws_access_key_id=settings.OOE_PATHSTREAM_S3_ACCESS_KEY,
+            aws_secret_access_key=settings.OOE_PATHSTREAM_S3_SECRET_KEY,
+        )
+
+    def _download_file(self):
+        """Download the file from an S3 Bucket.
+
+        :return: True if file was downloaded, else False"""
+        if self.client is None:
+            raise S3NotInitialized('_init_s3 has not been called yet.')
+
+        try:
+            self.client.download_file(
+                self.S3_BUCKET,
+                self.S3_FILE,
+                self.S3_FILE,
+            )
+        except ClientError as error:
+            LOG.error('Failed to download the file to S3. Reason: %s', str(error))
+            raise ExecutionError
+
+        LOG.info('File [%s] was downloaded from S3 for [%s]', self.S3_FILE, self.__str__())
+
+    def _update_file(self, enrollments_data):
+        """Append enrollment data to the downloaded S3 file
+
+        :param enrollments_data: List of the enrollments data
+        :return: True if file was updated, else False"""
```

Comment:
@Jacatove, does this method return something?

Reply:
No, it does not. I updated the docstring. @ivanvgh